rest/route/service.go

```diff
@@ -125,11 +125,12 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/patches/{patch_id}/abort").Version(2).Post().Wrap(checkUser, submitPatches).RouteHandler(makeAbortPatch(sc))
 	app.AddRoute("/patches/{patch_id}/restart").Version(2).Post().Wrap(checkUser, submitPatches).RouteHandler(makeRestartPatch(sc))
 	app.AddRoute("/projects").Version(2).Get().Wrap(checkUser).RouteHandler(makeFetchProjectsRoute(sc))
-	app.AddRoute("/projects/{project_id}").Version(2).Get().Wrap(checkUser, addProject, checkProjectAdmin, viewProjectSettings).RouteHandler(makeGetProjectByID(sc))
-	app.AddRoute("/projects/{project_id}").Version(2).Patch().Wrap(checkUser, addProject, checkProjectAdmin, editProjectSettings).RouteHandler(makePatchProjectByID(sc, env.Settings()))
+	app.AddRoute("/projects/{project_id}").Version(2).Get().Wrap(checkUser, checkProjectAdmin, viewProjectSettings).RouteHandler(makeGetProjectByID(sc))
+	app.AddRoute("/projects/{project_id}").Version(2).Patch().Wrap(checkUser, checkProjectAdmin, editProjectSettings).RouteHandler(makePatchProjectByID(sc, env.Settings()))
 	app.AddRoute("/projects/{project_id}").Version(2).Put().Wrap(superUser, createProject).RouteHandler(makePutProjectByID(sc))
-	app.AddRoute("/projects/{project_id}/copy").Version(2).Post().Wrap(checkUser, addProject, checkProjectAdmin, editProjectSettings).RouteHandler(makeCopyProject(sc))
-	app.AddRoute("/projects/{project_id}/events").Version(2).Get().Wrap(checkUser, addProject, checkProjectAdmin, viewProjectSettings).RouteHandler(makeFetchProjectEvents(sc))
+	app.AddRoute("/projects/{project_id}/copy").Version(2).Post().Wrap(checkUser, createProject, checkProjectAdmin, editProjectSettings).RouteHandler(makeCopyProject(sc))
+	app.AddRoute("/projects/{project_id}/copy_redacted").Version(2).Post().Wrap(checkUser, checkProjectAdmin, editProjectSettings).RouteHandler(makeCopyRedactedVars(sc))
```

Comment:
this should have variables in the name? this only the project variables and not the full project?

Reply:
I just wasn't sure if the name was getting too long / redacted does imply variables, I'm not attached to it though.